doc/Style.md

```diff
@@ -0,0 +1,248 @@
+# Agora coding style
+
+This document describes the style used in within Agora.
+It covers the style itself, which should be clear and unambiguous,
+as well as more design aspects on how we approach / solve problems.
+
+## Formatting
+
+We mostly follow [the D style](https://dlang.org/dstyle.html).
+
+The following is a list of exceptions:
+
+### Modules
+
+All module should have a module documentation. If you create a new module,
+just look at how another module is documented.
+
+Module names are CamelCase. While the D style recommend `std.exception`,
+we use `std.Exception`.
+This makes the module more prominent and disambiguate any symbol when
+reading fully qualified names, as the first CamelCase name is the module.
+
+Imports are organized by packages: `agora` first, dependencies second, `std` third,
+`core` last. Within packages, alphabetical order is prefered, with modules
+taking preference over packages, e.g.:
+```
+import agora.foo.Bar;
+import agora.foo.Zyx; // In alphabetical order, this would be last
+import agora.foo.pkg.Bar;
+```
+
+### Explicit, readable code
+
+Visibility attribute (`public`, `protected`, `package`, `private`) is always specified.
+Statement covering a scope without indentation (`private:`) should be avoided.
+This made code much easier to read, and code is read much more than it is written.
+
+When implementing an interface, while `override` is not enforced by the language,
+it is mandatory in our style, as changing the interface might lead to silent
+breakage of implementers without `override` being used.
+It also informs the reader that there is an additional context needed.
+
+Prefer longer, more descriptive names over short names in API (e.g. parameters),
+as it makes the documentation more readable.
+Likewise, when writing unittests for documentation purpose, avoid `auto` and
+`const` without a type spelled out.
+
+When importing outside modules, selective import (e.g. `import std.exception : enforce`)
+are prefered. This does not matter as much for `agora`, as modules tend to
+be smaller, with less dependencies.
+
+Put public symbols first in a module / aggregate whenever possible.
+
+### Functions
```

Comment:
There should also be a `Classes / Structs` section mentioning that fields should be placed before any methods.